### PR TITLE
BLOOM minor changes on tokenizer

### PR DESCRIPTION
**Pull request by younesbelkada**
# What does this PR do?

- Attempts to fix minor issues with the BloomTokenizer
- Remove unused args
- Added a new attribute on `TokenizerTesterMixin` for models that are not agnostic to sequence length (typically models that use ALiBi positional embeddings)

Still need to discuss if it is worth it to force the padding side to the left

cc @SaulLu

### CODE REVIEW

File: src/transformers/models/bloom/tokenization_bloom_fast.py [hunk 1]
```diff
@@ -109,7 +99,7 @@ class BloomTokenizerFast(PreTrainedTokenizerFast):
 
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
+    max_model_input_sizes = []
```
Comment (SaulLu) [inline src/transformers/models/bloom/tokenization_bloom_fast.py:102]: I am really in favor of this change, especially because with AliBi positional embeddings we don't have a max model input size (or pretrainned positional embeddings)

File: src/transformers/models/bloom/tokenization_bloom_fast.py [hunk 2]
```diff
@@ -123,6 +113,7 @@ def __init__(
         eos_token="</s>",
         pad_token="<pad>",
         add_prefix_space=False,
+        padding_side="left",
```
[the remark this answers is not in the document]
Comment (younesbelkada) [reply src/transformers/models/bloom/tokenization_bloom_fast.py:116]: Will remove it on the next commit, I have just confirmed that we produce the same result with different padding ;)

[the remark this answers is not in the document]
Comment (younesbelkada) [reply src/transformers/models/bloom/tokenization_bloom_fast.py]: Should be fixed in 261a7864dfdc2fe48b515d2ca950169030b61efa

File: src/transformers/models/bloom/tokenization_bloom_fast.py (hunk 1 above)
Comment (SaulLu) [reply src/transformers/models/bloom/tokenization_bloom_fast.py]: Your proposition looks good to me! :smile:

File: src/transformers/models/bloom/tokenization_bloom_fast.py [hunk 3]
```diff
@@ -109,7 +99,6 @@ class BloomTokenizerFast(PreTrainedTokenizerFast):
 
     vocab_files_names = VOCAB_FILES_NAMES
     pretrained_vocab_files_map = PRETRAINED_VOCAB_FILES_MAP
-    max_model_input_sizes = PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES
     model_input_names = ["input_ids", "attention_mask"]
     slow_tokenizer_class = None
```
Comment (SaulLu) [inline src/transformers/models/bloom/tokenization_bloom_fast.py:104]: WDYT of adding a comment here to keep a track in the code of the reason why this tokenizer doesn't have this field fill?
```suggestion
# No `max_model_input_sizes` as BLOOM uses ALiBi positional embeddings
```

**Comment by younesbelkada** [comment]
We may merge this at the same time as https://github.com/huggingface/transformers/pull/17837 to make it a patch release of minor fixes

**Review by SaulLu — APPROVED** [review]
All the changes look good to me! Thanks @younesbelkada :hugs:

**Review by sgugger — APPROVED** [review]
LGTM, thanks!
